src/polynomial.cpp

```diff
@@ -0,0 +1,310 @@
+#include "polynomial.h"
+#include "add.h"
+#include "mul.h"
+#include "pow.h"
+#include "constants.h"
+
+namespace SymEngine {
+
+    Polynomial::Polynomial(const RCP<const Symbol> &var, map_uint_mpz&& dict) :
```

Comment:
Rest of `SymEngine` code does not use four spaces like this before `Polynomial`. @certik, should this be changed?